packages/viewport/src/with-viewport-match.js

```diff
@@ -6,8 +6,69 @@ import { mapValues } from 'lodash';
 /**
  * WordPress dependencies
  */
+import { createContext, useContext } from '@wordpress/element';
 import { createHigherOrderComponent } from '@wordpress/compose';
-import { withSelect } from '@wordpress/data';
+import { useSelect } from '@wordpress/data';
+
+/**
+ * Hash of breakpoint names with pixel width at which it becomes effective.
+ *
+ * @see _breakpoints.scss
+ *
+ * @type {Object}
+ */
+export const BREAKPOINTS = {
+	huge: 1440,
+	wide: 1280,
+	large: 960,
+	medium: 782,
+	small: 600,
+	mobile: 480,
+};
+
+/**
+ * Hash of query operators with corresponding condition for media query.
+ *
+ * @type {Object}
+ */
+export const OPERATORS = {
+	'<': 'max-width',
+	'>=': 'min-width',
+};
+
+const OPERATOR_EVALUATORS = {
+	'<': ( breakpointValue, width ) => ( width < breakpointValue ),
+	'>=': ( breakpointValue, width ) => ( width >= breakpointValue ),
+};
+
+const ViewportMatchWidthContext = createContext( null );
+
+/**
+ * Component that makes all withViewportMatch usages descendents of its children,
+ * evaluate as if the viewport had the width specific as a prop.
+ */
+export const ViewportMatchWidthProvider = ViewportMatchWidthContext.Provider;
```

Comment:
Now that we have `useViewportMatch`, do you think we should move that provider to the compose package instead?

Reply:
Yes, I'm working on this. The uncertainty I'm finding is if we should also expose the consumer of the context to allow it to be used in simulations by withViewPortMatch. I wanted a single provider to simulate both in the hook and HOC.

Reply:
Another possibility is to refactor our existing class components that use the HoC to functional components and use the hook everywhere.

Reply:
It seems we still have lots of usages of withViewPortMatch and indirect usages via ifViewportMatch. I guess we can try to refactor all of those, but it will take some time. Another con of this solution is that some blocks may be using withViewPortMatch, and in that case, these use cases will not be simulated.
I guess we can simply expose the simulation via a hook: useViewportMatch.__experimentalUseSimulatedValue()

Then we can use the hook inside the withViewPortMatchSimulation to retrieve the simulated value from the context.

@youknowriad what you think of this idea?

Reply:
Not sure I follow though.

One viable approach too is to rewrite withViewportMatch to use useViewportMatch under the hood. that way we only have to support the simulation in the hook.

Reply:
Basically I was proposing an experimental hook that exposes the width simulation value from the context
```
const simulationWidth = seViewportMatch.__experimentalUseSimulatedValue();

if( simulationWidth ) {...
```

But I think making withViewPortMatch use the hook is a better idea and it is something we should do anyway because we need to keep withViewPortMatch for back-compatibility and doing this we converge to a single codebase (way of doing things).